packages/app/src/providers/h5grove/utils.ts

```diff
@@ -198,3 +204,24 @@ export function parseDType(type: H5GroveType): DType {
 
   return unknownType();
 }
+
+/*
+ * Take advantage of h5grove's "safe" dtype conversions (i.e. `dtype=safe`)
+ * to allow fetching more data as binary.
+ * https://github.com/silx-kit/h5grove/blob/3c851e748d52f5bb8eab12a7f8368781b86772da/h5grove/utils.py#L196
+ */
+export function h5groveTypedArrayFromDType(
+  dtype: DType,
+): TypedArrayConstructor | undefined {
+  const { class: dtypeClass } = dtype;
+
+  if (dtypeClass === DTypeClass.Float && dtype.size === 16) {
+    return Float32Array; // also saves a number[]->Float32Array conversion in texture-based visualizations
+  }
+
+  if (dtypeClass === DTypeClass.Float && dtype.size === 128) {
+    return Float64Array;
+  }
+
+  return undefined;
```

Comment:
It might be cleaner to `return typedArrayFromDType(type)` here to avoid the `||` in `h5grove-api.ts`.

Reply:
Ooh clever